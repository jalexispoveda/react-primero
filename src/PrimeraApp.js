import React from "react";
import PropTypes from "prop-types";

export default function PrimeraApp({ saludo, subtitulo }) {
  return (
    <>
      <h1>{saludo}</h1>
      <p>{subtitulo}</p>
    </>
  );
}

PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired,
};

PrimeraApp.defaultProps = {
  subtitulo: "Soy un subtitulo",
};
