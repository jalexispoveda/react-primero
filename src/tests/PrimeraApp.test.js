import { render } from "@testing-library/react";

import React from "react";
import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";

describe("Probando primeraApp component", () => {
  //   test("debe mostrar el mensaje hola soy goku", () => {
  //     const saludo = "Hola soy goku";
  //     //Producto renderizado de lo que yo voy a probar
  //     const wrapper = render(<PrimeraApp saludo={saludo}></PrimeraApp>);
  //     expect(wrapper.getByText(saludo)).toBeInTheDocument();
  //   });

  test("debe mostrar <primeraApp> component correctamente ", () => {
    const saludo = "Hola soy goku";
    const wrapper = shallow(<PrimeraApp saludo={saludo}></PrimeraApp>);
    //probando que el componente se muestr en pantalla correctamente

    expect(wrapper).toMatchSnapshot();
  });
});
