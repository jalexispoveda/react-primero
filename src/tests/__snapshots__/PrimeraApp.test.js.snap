// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando primeraApp component debe mostrar <primeraApp> component correctamente  1`] = `
<Fragment>
  <h1>
    Hola soy goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
